[bug]: unit flake: race in custodian from proof import

### Background

Unit test failure in the custodian tests from a CI run for #768:

```sh
--- FAIL: TestTransactionConfirmedOnly (4.77s)
    test_postgres.go:11: Creating new Postgres DB for testing
    custodian_test.go:220: 
        	Error Trace:	/home/runner/work/taproot-assets/taproot-assets/tapgarden/custodian_test.go:220
        	            				/home/runner/work/taproot-assets/taproot-assets/tapgarden/custodian_test.go:740
        	            				/home/runner/work/taproot-assets/taproot-assets/tapgarden/custodian_test.go:669
        	Error:      	Received unexpected error:
        	            	event 0 has status 1 but wanted 3
        	Test:       	TestTransactionConfirmedOnly
FAIL
FAIL	github.com/lightninglabs/taproot-assets/tapgarden	31.421s
FAIL
```



### Your environment

```sh
branch main
commit d37b48636515eaa5346bf07347fae8ec21f7a958
```

### Steps to reproduce

Running the race detector locally for the `tapgarden` tests:

`main unit-race pkg=tapgarden timeout=2m`

We see a failure in the same test:

```sh
WARNING: DATA RACE
Write at 0x00c0004e2500 by goroutine 3770:
  github.com/lightninglabs/taproot-assets/proof.(*MultiArchiver).ImportProofs.func1()
      /home/jhb/tap/review_repo/tap/proof/archive.go:811 +0x288
  github.com/lightninglabs/taproot-assets/fn.ParSlice[go.shape.*uint8].func1()
      /home/jhb/tap/review_repo/tap/fn/concurrency.go:29 +0x5d
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /home/jhb/go/pkg/mod/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:75 +0x76

Previous read at 0x00c0004e2500 by goroutine 3638:
  github.com/lightninglabs/taproot-assets/tapdb.(*AssetStore).importAssetFromProof()
      /home/jhb/tap/review_repo/tap/tapdb/assets_store.go:1464 +0x7ae
  github.com/lightninglabs/taproot-assets/tapdb.(*AssetStore).ImportProofs.func1()
      /home/jhb/tap/review_repo/tap/tapdb/assets_store.go:1601 +0x10d
  github.com/lightninglabs/taproot-assets/tapdb.(*TransactionExecutor[go.shape.interface { AnchorGenesisPoint(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.AnchorGenesisPointParams) error; ApplyPendingOutput(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.ApplyPendingOutputParams) (int64, error); ConfirmChainAnchorTx(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.ConfirmChainAnchorTxParams) error; DeleteAssetWitnesses(context.Context, int64) error; DeleteExpiredUTXOLeases(context.Context, database/sql.NullTime) error; DeleteManagedUTXO(context.Context, []uint8) error; DeleteUTXOLease(context.Context, []uint8) error; FetchAssetMetaByHash(context.Context, []uint8) (github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchAssetMetaByHashRow, error); FetchAssetMetaForAsset(context.Context, []uint8) (github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchAssetMetaForAssetRow, error); FetchAssetProof(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchAssetProofParams) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchAssetProofRow, error); FetchAssetProofs(context.Context) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchAssetProofsRow, error); FetchAssetProofsByAssetID(context.Context, []uint8) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchAssetProofsByAssetIDRow, error); FetchAssetWitnesses(context.Context, database/sql.NullInt64) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchAssetWitnessesRow, error); FetchChainTx(context.Context, []uint8) (github.com/lightninglabs/taproot-assets/tapdb/sqlc.ChainTxn, error); FetchGenesisID(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchGenesisIDParams) (int64, error); FetchGroupedAssets(context.Context) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchGroupedAssetsRow, error); FetchManagedUTXO(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchManagedUTXOParams) (github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchManagedUTXORow, error); FetchManagedUTXOs(context.Context) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchManagedUTXOsRow, error); FetchScriptKeyIDByTweakedKey(context.Context, []uint8) (int64, error); FetchTransferInputs(context.Context, int64) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchTransferInputsRow, error); FetchTransferOutputs(context.Context, int64) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.FetchTransferOutputsRow, error); HasAssetProof(context.Context, []uint8) (bool, error); InsertAssetTransfer(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.InsertAssetTransferParams) (int64, error); InsertAssetTransferInput(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.InsertAssetTransferInputParams) error; InsertAssetTransferOutput(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.InsertAssetTransferOutputParams) error; InsertNewAsset(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.InsertNewAssetParams) (int64, error); InsertPassiveAsset(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.InsertPassiveAssetParams) error; LogProofTransferAttempt(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.LogProofTransferAttemptParams) error; QueryAssetBalancesByAsset(context.Context, []uint8) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.QueryAssetBalancesByAssetRow, error); QueryAssetBalancesByGroup(context.Context, []uint8) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.QueryAssetBalancesByGroupRow, error); QueryAssetTransfers(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.QueryAssetTransfersParams) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.QueryAssetTransfersRow, error); QueryAssets(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.QueryAssetsParams) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.QueryAssetsRow, error); QueryPassiveAssets(context.Context, int64) ([]github.com/lightninglabs/taproot-assets/tapdb/sqlc.QueryPassiveAssetsRow, error); QueryProofTransferAttempts(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.QueryProofTransferAttemptsParams) ([]time.Time, error); ReAnchorPassiveAssets(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.ReAnchorPassiveAssetsParams) error; SetAssetSpent(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.SetAssetSpentParams) (int64, error); UpdateUTXOLease(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpdateUTXOLeaseParams) error; UpsertAssetGroupKey(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertAssetGroupKeyParams) (int64, error); UpsertAssetGroupWitness(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertAssetGroupWitnessParams) (int64, error); UpsertAssetMeta(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertAssetMetaParams) (int64, error); UpsertAssetProof(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertAssetProofParams) error; UpsertAssetProofByID(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertAssetProofByIDParams) error; UpsertAssetWitness(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertAssetWitnessParams) error; UpsertChainTx(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertChainTxParams) (int64, error); UpsertGenesisAsset(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertGenesisAssetParams) (int64, error); UpsertGenesisPoint(context.Context, []uint8) (int64, error); UpsertInternalKey(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertInternalKeyParams) (int64, error); UpsertManagedUTXO(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertManagedUTXOParams) (int64, error); UpsertScriptKey(context.Context, github.com/lightninglabs/taproot-assets/tapdb/sqlc.UpsertScriptKeyParams) (int64, error) }]).ExecTx()
      /home/jhb/tap/review_repo/tap/tapdb/interfaces.go:241 +0x2c8
  github.com/lightninglabs/taproot-assets/tapdb.(*TransactionExecutor[github.com/lightninglabs/taproot-assets/tapdb.ActiveAssetsStore]).ExecTx()
      /home/jhb/tap/review_repo/tap/tapdb/interfaces.go:205 +0x7c
  github.com/lightninglabs/taproot-assets/tapdb.(*AssetStore).ImportProofs()
      /home/jhb/tap/review_repo/tap/tapdb/assets_store.go:1592 +0x1ed
  github.com/lightninglabs/taproot-assets/proof.(*MultiArchiver).ImportProofs()
      /home/jhb/tap/review_repo/tap/proof/archive.go:844 +0x2bb
  github.com/lightninglabs/taproot-assets/tapgarden.(*Custodian).receiveProof()
      /home/jhb/tap/review_repo/tap/tapgarden/custodian.go:529 +0xbe2
  github.com/lightninglabs/taproot-assets/tapgarden.(*Custodian).watchInboundAssets.func2()
      /home/jhb/tap/review_repo/tap/tapgarden/custodian.go:293 +0x170

Goroutine 3770 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /home/jhb/go/pkg/mod/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:72 +0x124
  github.com/lightninglabs/taproot-assets/fn.ParSlice[go.shape.*uint8]()
      /home/jhb/tap/review_repo/tap/fn/concurrency.go:28 +0x10f
  github.com/lightninglabs/taproot-assets/proof.(*MultiArchiver).ImportProofs()
      /home/jhb/tap/review_repo/tap/proof/archive.go:836 +0x1af
  github.com/lightninglabs/taproot-assets/tapgarden.(*Custodian).receiveProof()
      /home/jhb/tap/review_repo/tap/tapgarden/custodian.go:529 +0xbe2
  github.com/lightninglabs/taproot-assets/tapgarden.(*Custodian).inspectWalletTx.func1()
      /home/jhb/tap/review_repo/tap/tapgarden/custodian.go:414 +0x186

Goroutine 3638 (running) created at:
  github.com/lightninglabs/taproot-assets/tapgarden.(*Custodian).watchInboundAssets()
      /home/jhb/tap/review_repo/tap/tapgarden/custodian.go:290 +0x9c4
  github.com/lightninglabs/taproot-assets/tapgarden.(*Custodian).Start.func1.1()
      /home/jhb/tap/review_repo/tap/tapgarden/custodian.go:166 +0x33
==================
--- FAIL: TestTransactionConfirmedOnly (2.14s)
    test_sqlite.go:11: Creating new SQLite DB handle for testing: /tmp/TestTransactionConfirmedOnly3967917637/001/tmp.db
    test_sqlite.go:11: Creating new SQLite DB handle for testing: /tmp/TestTransactionConfirmedOnly3967917637/002/tmp.db
    testing.go:1465: race detected during execution of test
FAIL
FAIL    github.com/lightninglabs/taproot-assets/tapgarden       9.953s
FAIL
```

It looks like we can end up with multiple goroutines calling `c.receiveProof()` at once, and operating on the same proofs. This can lead to issues with other behavior like updating event states.

--- a/Makefile
+++ b/Makefile
@@ -212,11 +212,15 @@ endif
 
 flakehunter: build-itest
 	@$(call print, "Flake hunting ${backend} integration tests.")
-	while [ $$? -eq 0 ]; do make itest-only; done
+	while [ $$? -eq 0 ]; do make itest-only-trace; done
 
 flake-unit:
 	@$(call print, "Flake hunting unit tests.")
-	while [ $$? -eq 0 ]; do '$(GOLIST) | $(XARGS) env $(GOTEST) -test.timeout=20m -count=1'; done
+	while [ $$? -eq 0 ]; do make unit nocache=1; done
+
+flake-unit-trace:
+	@$(call print, "Flake hunting unit tests in debug mode.")
+	while [ $$? -eq 0 ]; do make unit-trace nocache=1; done
 
 flake-unit-race:
 	@$(call print, "Flake hunting races in unit tests.")

--- a/itest/assertions.go
+++ b/itest/assertions.go
@@ -1198,6 +1198,34 @@ func AssertUniverseRootEquality(t *testing.T,
 	))
 }
 
+// AssertUniverseRootEqualityEventually checks that the universe roots returned
+// by two daemons are either equal eventually.
+func AssertUniverseRootEqualityEventually(t *testing.T,
+	clientA, clientB unirpc.UniverseClient) {
+
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultWaitTimeout)
+	defer cancel()
+
+	err := wait.NoError(func() error {
+		rootRequest := &unirpc.AssetRootRequest{}
+		universeRootsAlice, err := clientA.AssetRoots(ctxt, rootRequest)
+		require.NoError(t, err)
+		universeRootsBob, err := clientB.AssetRoots(ctxt, rootRequest)
+		require.NoError(t, err)
+
+		if !AssertUniverseRootsEqual(
+			universeRootsAlice, universeRootsBob,
+		) {
+
+			return fmt.Errorf("roots not equal")
+		}
+
+		return nil
+	}, defaultWaitTimeout)
+	require.NoError(t, err)
+}
+
 // AssertUniverseRoot makes sure the given universe root exists with the given
 // sum, either identified by the asset ID or group key.
 func AssertUniverseRoot(t *testing.T, client unirpc.UniverseClient,

--- a/itest/multi_asset_group_test.go
+++ b/itest/multi_asset_group_test.go
@@ -317,6 +317,10 @@ func testMultiAssetGroupSend(t *harnessTest) {
 		collectibleGroupMembers + 1,
 	})
 
+	AssertUniverseRootEqualityEventually(
+		t.t, t.tapd, t.universeServer.service,
+	)
+
 	// We'll make a second node now that'll be the receiver of all the
 	// assets made above.
 	secondTapd := setupTapdHarness(
@@ -325,6 +329,10 @@ func testMultiAssetGroupSend(t *harnessTest) {
 	defer func() {
 		require.NoError(t.t, secondTapd.stop(!*noDelete))
 	}()
+
+	AssertUniverseRootEqualityEventually(
+		t.t, secondTapd, t.universeServer.service,
+	)
 
 	// Send 5 of the assets to Bob, and verify that they are received.
 	numUnits := issuableAsset.Asset.Amount

--- a/proof/mock.go
+++ b/proof/mock.go
@@ -268,7 +268,29 @@ func (m *MockProofCourier) ReceiveProof(_ context.Context,
 		return nil, ErrProofNotFound
 	}
 
-	return proof, nil
+	return &AnnotatedProof{
+		Locator: Locator{
+			AssetID:   proof.Locator.AssetID,
+			GroupKey:  proof.Locator.GroupKey,
+			ScriptKey: proof.Locator.ScriptKey,
+			OutPoint:  proof.Locator.OutPoint,
+		},
+		Blob: proof.Blob,
+		AssetSnapshot: &AssetSnapshot{
+			Asset:             proof.AssetSnapshot.Asset,
+			OutPoint:          proof.AssetSnapshot.OutPoint,
+			AnchorBlockHash:   proof.AssetSnapshot.AnchorBlockHash,
+			AnchorBlockHeight: proof.AssetSnapshot.AnchorBlockHeight,
+			AnchorTxIndex:     proof.AssetSnapshot.AnchorTxIndex,
+			AnchorTx:          proof.AssetSnapshot.AnchorTx,
+			OutputIndex:       proof.AssetSnapshot.OutputIndex,
+			InternalKey:       proof.AssetSnapshot.InternalKey,
+			ScriptRoot:        proof.AssetSnapshot.ScriptRoot,
+			TapscriptSibling:  proof.AssetSnapshot.TapscriptSibling,
+			SplitAsset:        proof.AssetSnapshot.SplitAsset,
+			MetaReveal:        proof.AssetSnapshot.MetaReveal,
+		},
+	}, nil
 }
 
 // SetSubscribers sets the set of subscribers that will be notified

--- a/rpcserver.go
+++ b/rpcserver.go
@@ -3873,7 +3873,7 @@ func (r *rpcServer) QueryProof(ctx context.Context,
 	}
 
 	rpcsLog.Tracef("[QueryProof]: fetching proof at (universeID=%v, "+
-		"leafKey=%x)", universeID, leafKey.UniverseKey())
+		"leafKey=%x)", universeID.StringForLog(), leafKey.UniverseKey())
 
 	// Retrieve proof export config for the given universe.
 	syncConfigs, err := r.cfg.UniverseFederation.QuerySyncConfigs(ctx)
@@ -3937,7 +3937,8 @@ func (r *rpcServer) QueryProof(ctx context.Context,
 
 			rpcsLog.Debugf("[QueryProof]: error querying for "+
 				"proof at (universeID=%v, leafKey=%x)",
-				universeID, leafKey.UniverseKey())
+				universeID.StringForLog(),
+				leafKey.UniverseKey())
 			return nil, err
 		}
 
@@ -3953,12 +3954,13 @@ func (r *rpcServer) QueryProof(ctx context.Context,
 
 	// TODO(roasbeef): query may return multiple proofs, if allow key to
 	// not be fully specified
-	proof := proofs[0]
+	firstProof := proofs[0]
 
 	rpcsLog.Tracef("[QueryProof]: found proof at (universeID=%v, "+
-		"leafKey=%x)", universeID, leafKey.UniverseKey())
+		"leafKey=%x)", universeID.StringForLog(),
+		leafKey.UniverseKey())
 
-	return r.marshalUniverseProofLeaf(ctx, req, proof)
+	return r.marshalUniverseProofLeaf(ctx, req, firstProof)
 }
 
 // unmarshalAssetLeaf unmarshals an asset leaf from the RPC form.

--- a/tapdb/asset_minting.go
+++ b/tapdb/asset_minting.go
@@ -1096,10 +1096,11 @@ func (a *AssetMintingStore) CommitSignedGenesisTx(ctx context.Context,
 			AnchorUtxoID: sqlInt64(utxoID),
 		})
 		if err != nil {
-			return fmt.Errorf("unable to anchor pending assets: %w", err)
+			return fmt.Errorf("unable to anchor pending assets: %w",
+				err)
 		}
 
-		// Next, we'll anchor the genesis point to point to the chain
+		// Next, we'll anchor the genesis point-to-point to the chain
 		// transaction we inserted above.
 		if err := q.AnchorGenesisPoint(ctx, GenesisPointAnchor{
 			PrevOut:    genesisOutpoint,

--- a/tapdb/asset_minting_test.go
+++ b/tapdb/asset_minting_test.go
@@ -244,7 +244,7 @@ func TestCommitMintingBatchSeedlings(t *testing.T) {
 	mintingBatch := tapgarden.RandSeedlingMintingBatch(t, numSeedlings)
 	addRandGroupToBatch(t, assetStore, ctx, mintingBatch.Seedlings)
 	err := assetStore.CommitMintingBatch(ctx, mintingBatch)
-	require.NoError(t, err, "unable to write batch: %v", err)
+	require.NoError(t, err)
 
 	batchKey := mintingBatch.BatchKey.PubKey
 
@@ -267,11 +267,9 @@ func TestCommitMintingBatchSeedlings(t *testing.T) {
 	// Pick a random seedling and give it a specific group.
 	addRandGroupToBatch(t, assetStore, ctx, seedlings)
 	mintingBatch.Seedlings = mergeMap(mintingBatch.Seedlings, seedlings)
-	require.NoError(t,
-		assetStore.AddSeedlingsToBatch(
-			ctx, batchKey, maps.Values(seedlings)...,
-		), "unable to write seedlings: %v", err,
-	)
+	require.NoError(t, assetStore.AddSeedlingsToBatch(
+		ctx, batchKey, maps.Values(seedlings)...,
+	))
 
 	// If we read the batch from disk again, then we should have 10 total
 	// seedlings, and the batch still matches what we wrote to disk.
@@ -1088,7 +1086,7 @@ func TestGroupAnchors(t *testing.T) {
 	)
 	addMultiAssetGroupToBatch(t, mintingBatch.Seedlings)
 	err := assetStore.CommitMintingBatch(ctx, mintingBatch)
-	require.NoError(t, err, "unable to write batch: %v", err)
+	require.NoError(t, err)
 
 	batchKey := mintingBatch.BatchKey.PubKey
 

--- a/tapdb/multiverse.go
+++ b/tapdb/multiverse.go
@@ -538,7 +538,7 @@ func namespaceForProof(proofType universe.ProofType) (string, error) {
 		return transferMultiverseNS, nil
 
 	default:
-		return "", fmt.Errorf("unknown proof type: %v", int(proofType))
+		return "", fmt.Errorf("unknown proof type: %d", int(proofType))
 	}
 }
 

--- a/tapdb/sqlutils.go
+++ b/tapdb/sqlutils.go
@@ -167,7 +167,7 @@ func parseCoalesceNumericType[T constraints.Integer](value any) (T, error) {
 		parsedValue, err := strconv.ParseInt(typedValue, 10, 64)
 		if err != nil {
 			return 0, fmt.Errorf("unable to parse value '%v' as "+
-				"number: %v", value, err)
+				"number: %w", value, err)
 		}
 
 		return T(parsedValue), nil

--- a/tapgarden/custodian.go
+++ b/tapgarden/custodian.go
@@ -284,6 +284,14 @@ func (c *Custodian) watchInboundAssets() {
 			continue
 		}
 
+		// If this event is not yet confirmed, we don't yet expect a
+		// proof to be delivered. We'll wait for the confirmation to
+		// come in, and then we'll launch a goroutine to use the
+		// ProofCourier to import the proof into our local DB.
+		if event.ConfirmationHeight == 0 {
+			continue
+		}
+
 		// If we didn't find a proof, we'll launch a goroutine to use
 		// the ProofCourier to import the proof into our local DB.
 		c.Wg.Add(1)
@@ -530,7 +538,8 @@ func (c *Custodian) receiveProof(addr *address.Tap, op wire.OutPoint) error {
 		ctx, headerVerifier, c.cfg.GroupVerifier, false, addrProof,
 	)
 	if err != nil {
-		return fmt.Errorf("unable to import proofs: %w", err)
+		return fmt.Errorf("unable to import proofs script_key=%x, "+
+			"asset_id=%x: %w", scriptKeyBytes, assetID[:], err)
 	}
 
 	// The proof is now verified and in our local archive. We will now

--- a/tapgarden/planter_test.go
+++ b/tapgarden/planter_test.go
@@ -11,8 +11,6 @@ import (
 	"testing"
 	"time"
 
-	"github.com/lightninglabs/taproot-assets/tapsend"
-
 	"github.com/btcsuite/btcd/blockchain"
 	"github.com/btcsuite/btcd/btcec/v2"
 	"github.com/btcsuite/btcd/btcutil"
@@ -30,6 +28,7 @@ import (
 	_ "github.com/lightninglabs/taproot-assets/tapdb" // Register relevant drivers.
 	"github.com/lightninglabs/taproot-assets/tapgarden"
 	"github.com/lightninglabs/taproot-assets/tapscript"
+	"github.com/lightninglabs/taproot-assets/tapsend"
 	"github.com/lightningnetwork/lnd/build"
 	"github.com/lightningnetwork/lnd/keychain"
 	"github.com/lightningnetwork/lnd/lntest/wait"

